openmdao/core/system.py

```diff
@@ -835,26 +861,33 @@ def _setup_vectors(self, root_vectors, excl_out, excl_in, resize=False):
             Dictionary of sets of excluded input variable absolute names, keyed by vec_name.
         resize : bool
             Whether to resize the root vectors - i.e, because this system is initiating a reconf.
+        alloc_complex : bool
+            Whether to allocate any imaginary storage to perform complex step. Default is False.
         """
         self._vectors = vectors = {'input': {}, 'output': {}, 'residual': {}}
         self._excluded_vars_out = excl_out
         self._excluded_vars_in = excl_in
 
+        # Check for complex step to set vectors up appropriately.
+        if 'cs' in self._approx_schemes or alloc_complex:
```

Comment:
I say we force the user to declare alloc_complex before hand, and if they don't but 'cs' is in approx_schemes, we should throw an error. Otherwise, we may run into ugly situations with reconfigurability in which a subgroup or subcomponent adds a new CS approximation, where the root vector is not declared to be complex